File: Hashmap/Valid Anagram/ValidAnagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }
        
        Dictionary<char, int> counter1 = new Dictionary<char, int>();
        Dictionary<char, int> counter2 = new Dictionary<char, int>();
        
        foreach (char c in s) {
            if (counter1.ContainsKey(c)) {
                counter1[c]++;
            } else {
                counter1[c] = 1;
            }
        }
        
        foreach (char c in t) {
            if (counter2.ContainsKey(c)) {
                counter2[c]++;
            } else {
                counter2[c] = 1;
            }
        }
        
        return counter1.OrderBy(kv => kv.Key)
                       .SequenceEqual(counter2.OrderBy(kv => kv.Key));
    }
}

File: Hashmap/Valid Anagram/ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        map<char, int> counter1, counter2;
        for(char c : s) {
            counter1[c]++;
        }
        for(char c : t) {
            counter2[c]++;
        }
        return counter1 == counter2;
    }
};